### ISSUE TITLE
input quantity doesn't work - Uncaught TypeError

### ISSUE THREAD

**Issue by noisysomeone** [wojodesign/simplecart-js#123]
Hi,
I have some problem with my simpleCart. I get error:"Uncaught TypeError: Object 5 has no method 'match'" in line:895 when i tried to type quantity in input. The same problem is on: http://demo.simplecartjs.com/ Try to write quantity in input field- it doesn't work.

I have 2.2 version. Browser: firefox, chrome. Thank you for help:)

**Comment by brettwejrowski** [comment]
Hi noisysomeone, thanks so much for pointing this out!  I figured out what you were talking about, and I added some tests for it and fixed the bug in both the master branch and 2.3.dev.  

You should be all set now. Thanks!

**Comment by brettwejrowski** [comment]
sure. you need to change the "set" function (first function of the CartItem prototype, on line 937 to this:

``` javascript
set : function ( field , value ){
        field = field.toLowerCase();
        if( typeof( this[field] ) !== "function" && field !== "id" ){
            value = "" + value;
            if( field == "quantity"){
                value = value.replace( /[^(\d|\.)]*/gi , "" );
                value = value.replace(/,*/gi, "");
                value = parseInt(value,10);
            } else if( field == "price" ){
                value = value.replace( /[^(\d|\.)]*/gi, "");
                value = value.replace(/,*/gi , "");
                value = parseFloat( value );
            }
            if( typeof(value) == "number" && isNaN( value ) ){
                error( "Improperly formatted input.");
            } else {
                if( typeof( value ) === "string" ){
                    if( value.match(/\~|\=/) ){
                        error("Special character ~ or = not allowed: " + value);
                    }
                    value = value.replace(/\~|\=/g, "");
                }
                this[field] = value;
                this.checkQuantityAndPrice();
            }
        } else {
            error( "Cannot change " + field + ", this is a reserved field.");
        }
        simpleCart.update();
    },
```
